### PR TITLE
Exposure Submission Flow - Error Alert adjustments

### PR DESCRIPTION
**Pull request by melloskitten**
### Description

This PR aims to implement a better error handling of the infamous EN errors, as described in this issue [here](https://jira.itc.sap.com/browse/EXPOSUREAPP-1552). **Please note that this handles the errors thrown in the Exposure Submission flow primarily.**

It will handle the ENError 5, 11, and 13. All of them now have an Alert Action "Mehr erfahren" that leads you to the FAQ.

I have also refactored the Error handling for the Exposure Submission flow, which had a lot of unecessary methods that did exactly the same thing. I also adjusted some namings because they seemed confusing to me (e.g. retry -> secondaryAction for alerts, and so on.)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionWarnOthersViewController.swift [hunk 1]
```diff
@@ -53,18 +53,26 @@ class ExposureSubmissionWarnOthersViewController: DynamicTableViewController, EN
 		exposureSubmissionService = exposureSubmissionService ?? (navigationController as? ExposureSubmissionNavigationController)?.exposureSubmissionService
 	}
 
-	// MARK: - ExposureSubmissionService Helpers.
```
Comment (MichaelPFrey) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionWarnOthersViewController.swift:56]: why do you remove this MARK?

Comment (melloskitten) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionWarnOthersViewController.swift:56]: Ah, thanks for the comment. Accident on my side 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ErrorAlertUtil.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	///   - title: The title of the alert. If omitted, it will use the general error title.
+	///   - message: The description of the alert.
+	///   - okTitle: The text of the ok action.
+	///   - secondaryActionTitle: The text of the secondary action, if there is one.
+	///   - hasSecondaryAction: Indicates whether an alert has a secondary action or not.
+	///   - completion: The completion handler for the "ok" action.
+	///   - secondaryActionCompletion: The completion handler for the secondary action.
+	/// - Returns: An alert with either one or two actions, with the specified completion handlers
+	/// and texts.
+	static func setupErrorAlert(
```
Comment (steins-code) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ErrorAlertUtil.swift:34]: Do we really need this additional file? Or can we consolidate it and use only extension of UIViewController + Alert.swift and maybe extend this one

Comment (melloskitten) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ErrorAlertUtil.swift]: Good point. I have refactored it into the already existing extension 👍

File: src/xcode/ENA/ENA/Source/Extensions/UIViewController+Alert.swift [hunk 3]
```diff
@@ -24,4 +24,52 @@ extension UIViewController {
...
+	///   - completion: The completion handler for the "ok" action.
+	///   - secondaryActionCompletion: The completion handler for the secondary action.
+	/// - Returns: An alert with either one or two actions, with the specified completion handlers
+	/// and texts.
+	public func setupErrorAlert(
+		title: String? = nil,
+		message: String,
+		okTitle: String? = nil,
+		secondaryActionTitle: String? = nil,
+		hasSecondaryAction: Bool = false, completion: (() -> Void)? = nil,
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Extensions/UIViewController+Alert.swift:44]: missing line break

**Review by johannesrohwer — APPROVED** [review]
looks good to me

File: src/xcode/ENA/ENA/Source/Extensions/UIViewController+Alert.swift (hunk 3 above)
Comment (melloskitten) [reply src/xcode/ENA/ENA/Source/Extensions/UIViewController+Alert.swift]: done, thanks for the notice.

**Review by steins-code — APPROVED** [review]

**Pull request merged**